src/protocol/get_host_state.rs

```diff
@@ -0,0 +1,129 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+//! `GetHostState` request and response.
+//!
+//! This module provides a Cerberus command that allows the querying of
+//! the reset state of the host processor protected by Cerberus.
+
+use crate::io::Read;
+use crate::io::Write;
+use crate::mem::Arena;
+use crate::protocol::wire::FromWire;
+use crate::protocol::wire::FromWireError;
+use crate::protocol::wire::ToWire;
+use crate::protocol::wire::ToWireError;
+use crate::protocol::Command;
+use crate::protocol::CommandType;
+use crate::protocol::Request;
+use crate::protocol::Response;
+
+#[cfg(feature = "arbitrary-derive")]
+use libfuzzer_sys::arbitrary::{self, Arbitrary};
+#[cfg(feature = "serde")]
+use serde::{Deserialize, Serialize};
+
+/// A command for requesting the host reset state
+///
+/// Corresponds to [`CommandType::GetHostState`].
+pub enum GetHostState {}
+
+impl<'a> Command<'a> for GetHostState {
+    type Req = GetHostStateRequest;
+    type Resp = GetHostStateResponse;
+}
+
+/// The [`GetHostState`] request.
+#[derive(Clone, Copy, PartialEq, Eq, Debug)]
+#[cfg_attr(feature = "arbitrary-derive", derive(Arbitrary))]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+pub struct GetHostStateRequest {
+    port_id: u8,
+}
+make_fuzz_safe!(GetHostStateRequest);
+
+impl Request<'_> for GetHostStateRequest {
+    const TYPE: CommandType = CommandType::GetHostState;
+}
+
+impl<'a> FromWire<'a> for GetHostStateRequest {
+    fn from_wire<R: Read, A: Arena>(
+        mut r: R,
+        _: &'a A,
+    ) -> Result<Self, FromWireError> {
+        let port_id = r.read_le()?;
+        Ok(Self { port_id })
+    }
+}
+
+impl<'a> ToWire for GetHostStateRequest {
+    fn to_wire<W: Write>(&self, mut w: W) -> Result<(), ToWireError> {
+        w.write_le(self.port_id)?;
+        Ok(())
+    }
+}
+
+make_fuzz_safe! {
+    /// The [`GetHostState`] response.
+    #[derive(Clone, Copy, PartialEq, Eq, Debug)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
```

Comment:
Uh oh, looks like `rustfmt` got confused. I bet you copy-pasted this from a place where I got this wrong, hehe.

Reply:
Fixed. I had thought some macro parsing oddity earlier, but turns out I was misreading the compiler error I was fixing at the time.